Foundation/NSOrderedSet.swift

```diff
@@ -37,38 +48,151 @@ public class NSOrderedSet : NSObject, NSCopying, NSMutableCopying, NSSecureCodin
     
     public required init?(coder aDecoder: NSCoder) { NSUnimplemented() }
     
-    public var count: Int { NSUnimplemented() }
-    public func objectAtIndex(idx: Int) -> AnyObject { NSUnimplemented() }
-    public func indexOfObject(object: AnyObject) -> Int { NSUnimplemented() }
-    public override init() { NSUnimplemented() }
-    public init(objects: UnsafePointer<AnyObject?>, count cnt: Int) { NSUnimplemented() }
+    public var count: Int {
+        return _entries.count
+    }
+
+    public func objectAtIndex(idx: Int) -> AnyObject {
+        return _entryAtIndex(idx).object
```

Comment:
Original implementation of the `NSOrderedSet`/`NSMutableOrderedSet` uses array and set internally (WWDC 2013, Session 224). This doubles memory usage, but make it possible to get the best of both worlds.